File: IOTConnectSDK/app/src/main/java/com/softweb/iotconnectsdk/model/SdkOptions.java
package com.softweb.iotconnectsdk.model;

import com.google.gson.annotations.SerializedName;

public class SdkOptions {

    @SerializedName("offlineStorage")
    private OfflineStorage offlineStorage;

    @SerializedName("certificate")
    private Certificate certificate;

    @SerializedName("discoveryUrl")
    private String discoveryUrl;

    @SerializedName("isDebug")
    private boolean isDebug = false;

    @SerializedName("skipValidation")
    private boolean skipValidation = false;

    public OfflineStorage getOfflineStorage() {
        return offlineStorage;
    }

    public Certificate getCertificate() {
        return certificate;
    }

    public String getDiscoveryUrl() {
        return discoveryUrl;
    }

    public void setDiscoveryUrl(String discoveryUrl) {
        this.discoveryUrl = discoveryUrl;
    }

    public void setOfflineStorage(OfflineStorage offlineStorage) {
        this.offlineStorage = offlineStorage;
    }

    public void setCertificate(Certificate certificate) {
        this.certificate = certificate;
    }

    public boolean isDebug() {
        return isDebug;
    }

    public void setDebug(boolean debug) {
        isDebug = debug;
    }


    public boolean isSkipValidation() {
        return skipValidation;
    }

    public void setSkipValidation(boolean skipValidation) {
        this.skipValidation = skipValidation;
    }
}

File: IOTConnectSDK/iotconnect-sdk_2_1/src/main/java/com/iotconnectsdk/iotconnectconfigs/SdkOptions.kt
package com.iotconnectsdk.iotconnectconfigs

import com.google.gson.annotations.SerializedName
import com.iotconnectsdk.IoTCEnvironment
import com.iotconnectsdk.enums.BrokerType

class SdkOptions {
    @JvmField
    @SerializedName("offlineStorage")
    var offlineStorage: OfflineStorage? = null

    @JvmField
    @SerializedName("certificate")
    var certificate: Certificate? = null

    @JvmField
    @SerializedName("discoveryUrl")
    var discoveryUrl: String? = null

    @JvmField
    @SerializedName("isDebug")
    var isDebug = false

    @JvmField
    @SerializedName("skipValidation")
    var isSkipValidation = false

    @JvmField
    @SerializedName("devicePK")
    var devicePK: String? = null

    @JvmField
    @SerializedName("cpId")
    var cpId = ""

    @JvmField
    @SerializedName("env")
    var env: IoTCEnvironment? = null

    @JvmField
    @SerializedName("pf")
    var pf = ""

}